src/v2/controllers/RegistrationFormController.js

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) 2020, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import FormController from './FormController';
+import { FORMS } from '../ion/RemediationConstants';
+
+export default FormController.extend({
+  postRender() {
+    // TODO: what if user navigate to this page but there is valid state token
+    // in session storage
+    // - try to sign-in
+    // - during at middle of sign-in, change URL to /signin/register/
+    // user probably expects to see sign-up page
+    // actually widget will continue show sign-in flow (because of the saved state token)
+    if (this.options.appState.get('currentFormName') === FORMS.IDENTIFY) {
+      // 1. auto switch to porfile enroll (register) page
+      // basically mimic the flow that show identify page and click sign-up link.
+      if (this.options.appState.hasRemediationObject(FORMS.SELECT_ENROLL_PROFILE)) {
+        this.handleInvokeAction(FORMS.SELECT_ENROLL_PROFILE);
+      } else {
+        const messages = {
+          'type': 'array',
+          value: [
+            {
+              message: 'registration is not supported',
+              class: 'ERROR',
+            }
+          ]
+        };
+
+        const resp = {
+          neededToProceed: [],
+          // OKTA-382410 so bad that has to leverage rawIdxState
+          rawIdxState: {
+            messages
+          },
+          context: {
+            messages
+          }
+        };
+        // this.options.appState.setIonResponse(resp);
+        this.options.appState.trigger('remediationSuccess', resp);
```

Comment:
When we show them the error in the terminal state `registration is not supported` should we have a link for them to go back to the login page ?

Reply:
Good point!

Reply:
should we even show this error message to enduser ? It will not make sense to them..  

I think user should not even see the page or link, if it is not enabled at the first place. Like pre-validation and avoid this state somehow.

Reply:
Assume you meant default login page where server side can valid and return 40x.
How we gonna avoid this for customer hosted page where we have no control of server but only the widget?

Reply:
Instead of showing this error we can redirect user to some default page with warning like `Registration form is not enabled for this org`

and show them default login page.. 

I am inclining toward not showing errors (specially for those , for which enduser's will scratch their heads) like `Invalid state token provided`..

Reply:
Add another commit to customize the message and add `back to sign-in` link.